sympy/integrals/risch.py

```diff
@@ -772,7 +772,7 @@ def gcdex_diophantine(a, b, c):
     q = c.exquo(g)  # Inexact division means c is not in (a, b)
     s = q*s
 
-    if not s.is_zero and b.degree() >= b.degree():
+    if not s.is_zero and s.degree() >= b.degree():
```

Comment:
What is the effect of this change? I tried the added test and it seems to pass on master anyway. Do you have a test that hits the line above?

Reply:
That was just a typo I came across in the codebase. There was only one test related to `gcdex_diophantine`, so I added another.  The test that I have added hits the above line:
```
>>> b
Poly(x**3 + 1, x, domain='ZZ')
>>> s
Poly(-x**3 - x**2 + x, x, domain='QQ')
>>> not s.is_zero and s.degree() >= b.degree()
True
```

Reply:
I don't have any test for which this `if` statement returns false. `b.degree() == b.degree()` always and `s.is_zero` is False in most cases.

Reply:
Since this branch is untested it would be good to invent a test and check that the output is correct.

Reply:
FYI, the test that I've already added with the PR was taken from [wikipedia](https://en.wikipedia.org/wiki/Polynomial_Diophantine_equation). And the output from our logic matches with the result mentioned on Wikipedia.

Reply:
@oscarbenjamin I think it's good to go now :)

Reply:
Previously this was `b.degree() >= b.degree()` which is always True since both sides are equal.

Now we have `s.degree() >= b.degree()` which can be False. Are there any tests for which that condition would be False?

Reply:
I'm sorry I misunderstood. 
> Are there any tests for which that condition would be False?

Here is an example from [StackExchange](https://math.stackexchange.com/questions/2472142/euclidean-algorithm-for-polynomials),
```
>>> a = Poly(x**3 + 6*x + 7)
>>> b = Poly(x**2 + 3*x + 2)
>>> c = Poly(x + 1)
>>> s
Poly(1/13, x, domain='QQ')
>>> b
Poly(x**2 + 3*x + 2, x, domain='ZZ')
>>> gcdex_diophantine(Poly(x**3 + 6*x + 7), Poly(x**2 + 3*x + 2), Poly(x + 1))
(Poly(1/13, x, domain='QQ'), Poly(-1/13*x + 3/13, x, domain='QQ'))
>>> s.degree() >= b.degree() 
False
```
Our output is correct.

Reply:
Is the output the same on master as it is with this PR?

Reply:
Yes, the output is same. But I fixed this typo just because of this condition mentioned [here](https://github.com/sympy/sympy/blob/fa1432db689eb5eb49439d372a98f29a82627ba6/sympy/integrals/risch.py#L764).

Reply:
I think that the degree test is there only to find out if the division of polynomials could be skipped.

Reply:
> Clearly we do not have a test for that case so a test should be added.

No idea about that. Please suggest if you have one @jksuom

Reply:
> I think that the degree test is there only to find out if the division of polynomials could be skipped.

Yes, so maybe it won't create any difference.

Reply:
Simple polynomials like

    a, b, c = Poly(x**2 - 1), Poly(x**3 - 1), Poly(x - 1)

could be added to the existing test https://github.com/sympy/sympy/blob/1b9fffb6bd729169caebb7498da1d6693c95e581/sympy/integrals/tests/test_risch.py#L19-L22

Reply:
So adding the stackexchange example should be sufficient to provide a case where the `if` logic would be skipped and the right answer still obtained.

Reply:
@smichr @jksuom Thanks for the suggestions. I'll update now.